contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol

```diff
@@ -0,0 +1,388 @@
+
+    /// @notice Set milestones for recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestones The milestones to be set
+    function setMilestones(address _recipientId, Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
+        if (upcomingMilestone[_recipientId] != 0) {
+            revert MILESTONES_ALREADY_SET();
+        }
+
+        uint256 totalAmountPercentage;
+
+        uint256 milestonesLength = _milestones.length;
+        for (uint256 i = 0; i < milestonesLength;) {
+            totalAmountPercentage += _milestones[i].amountPercentage;
+            milestones[_recipientId].push(_milestones[i]);
+
+            unchecked {
+                i++;
+            }
+        }
+
+        if (totalAmountPercentage != 1e18) {
+            revert INVALID_MILESTONE();
+        }
+
+        emit MILESTONES_SET(_recipientId);
+    }
+
+    /// @notice Submit milestone by the recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _metadata The proof of work
+    function submitMilestone(address _recipientId, Metadata calldata _metadata) external {
+        if (_recipientId != msg.sender && !_isIdentityMember(_recipientId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        uint256 milestoneToBeSubmitted = upcomingMilestone[_recipientId];
+        Milestone[] storage recipientMilestones = milestones[_recipientId];
+
+        if (milestoneToBeSubmitted > recipientMilestones.length) {
+            revert INVALID_MILESTONE();
+        }
+
+        Milestone storage milestone = recipientMilestones[milestoneToBeSubmitted];
+        milestone.metadata = _metadata;
+        milestone.milestoneStatus = RecipientStatus.Pending;
+
+        emit MILESTONE_SUBMITTED(_recipientId, milestoneToBeSubmitted, _metadata);
+    }
+
+    /// @notice Reject pending milestone of the recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestoneId Id of the milestone
+    function rejectMilestone(address _recipientId, uint256 _milestoneId) external onlyPoolManager(msg.sender) {
+        Milestone storage milestone = milestones[_recipientId][_milestoneId];
+        if (milestone.milestoneStatus == RecipientStatus.Accepted) {
+            revert MILESTONE_ALREADY_ACCEPTED();
+        }
+
+        milestone.milestoneStatus = RecipientStatus.Rejected;
+        emit MILESTONE_REJECTED(_recipientId, _milestoneId);
+    }
+
+    /// @notice Set the internal status of the recipient to InReview
+    /// @param _recipientIds Ids of the recipients
+    function setIntenalRecipientStatusToInReview(address[] calldata _recipientIds)
+        external
+        onlyPoolManager(msg.sender)
+    {
+        uint256 recipientLength = _recipientIds.length;
+        for (uint256 i = 0; i < recipientLength;) {
+            _recipients[_recipientIds[i]].recipientStatus = InternalRecipientStatus.InReview;
+            unchecked {
+                i++;
+            }
```

Comment:
Should we emit an event on status change?
General question for all of our strategies @thelostone-mc @codenamejason

Reply:
Yes we should